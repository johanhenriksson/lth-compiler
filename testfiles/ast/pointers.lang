int main() {
    int a = 5;
    int* a_ptr = &a;
    return *a_ptr;
}
