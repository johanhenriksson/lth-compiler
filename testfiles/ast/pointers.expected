Program 
  List 1 item(s)
    Function main()
      IdDecl int main
        TypeId int
      List 0 item(s)
      Block 3 statement(s)
        List 3 item(s)
          DeclareStmt int a = 5;
            IdDecl int a
              TypeId int
            Opt ?
              IntLiteral 5
          DeclareStmt int a_ptr = a;
            IdDecl int a_ptr
              PointerTypeId int
            Opt ?
              AddressExpr a
                IdExpr a
                  IdUse a
          ReturnStmt return a_ptr;
            Opt ?
              DerefExpr a_ptr
                IdExpr a_ptr
                  IdUse a_ptr
